### PR TITLE
#VP-3600 integrate security via GraphQL

### PR DESCRIPTION
**Pull request by kostyrin**

### CODE REVIEW

File: VirtoCommerce.Storefront.Model/Security/Contracts/DeleteUserResponseDto.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+namespace VirtoCommerce.Storefront.Model.Security.Contracts
+{
+    public class DeleteUserResponseDto
+    {
+        public bool DeleteUser { get; set; }
```
Comment (eMazeika) [inline VirtoCommerce.Storefront.Model/Security/Contracts/DeleteUserResponseDto.cs:5]: 1. the returned type is SecurityResultDto, not bool
2. no need for dedicated ResponseDto class in here. Create and use more general SecurityResponseDto

File: VirtoCommerce.Storefront.Model/Security/Contracts/UpdateRoleResponseDto.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+namespace VirtoCommerce.Storefront.Model.Security.Contracts
+{
+    public class UpdateRoleResponseDto
```
Comment (eMazeika) [inline VirtoCommerce.Storefront.Model/Security/Contracts/UpdateRoleResponseDto.cs:3]: no need for dedicated ResponseDto class in here. Create and use more general SecurityResponseDto

File: VirtoCommerce.Storefront.Model/Security/Contracts/UpdateUserResponseDto.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace VirtoCommerce.Storefront.Model.Security.Contracts
+{
+    public class UpdateUserResponseDto
```
Comment (eMazeika) [inline VirtoCommerce.Storefront.Model/Security/Contracts/UpdateUserResponseDto.cs:3]: no need for dedicated ResponseDto class in here. Create and use more general SecurityResponseDto

File: VirtoCommerce.Storefront/Domain/Security/GraphQlSecurityHelper.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
+namespace VirtoCommerce.Storefront.Domain.Security
+{
+    public static class GraphQlSecurityHelper
+    {
+        public static readonly string SecurityResultFields = $"succeeded errors";
```
Comment (eMazeika) [inline VirtoCommerce.Storefront/Domain/Security/GraphQlSecurityHelper.cs:5]: check errors structure

File: VirtoCommerce.Storefront/Domain/Security/GraphQlSecurityHelper.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          securityResult: updateUser(command: $command)
+          {{
+            { selectedFields ?? SecurityResultFields }
+          }}
+        }}";
+
+        public static string DeleteUserRequest(this SecurityGraphQLProvider service)
+        => $@"mutation ($command: InputDeleteUserType!)
+        {{
+            deleteContact(command: $command)
```
Comment (eMazeika) [inline VirtoCommerce.Storefront/Domain/Security/GraphQlSecurityHelper.cs:32]: add alias

File: VirtoCommerce.Storefront.Model/Security/Contracts/UpdateUserResponseDto.cs (hunk 3 above)
Comment (kostyrin) [reply VirtoCommerce.Storefront.Model/Security/Contracts/UpdateUserResponseDto.cs:3]: image

File: VirtoCommerce.Storefront.Model/Security/Contracts/UpdateRoleResponseDto.cs (hunk 2 above)
Comment (kostyrin) [reply VirtoCommerce.Storefront.Model/Security/Contracts/UpdateRoleResponseDto.cs:3]: look at down

File: VirtoCommerce.Storefront.Model/Security/Contracts/DeleteUserResponseDto.cs (hunk 1 above)
Comment (kostyrin) [reply VirtoCommerce.Storefront.Model/Security/Contracts/DeleteUserResponseDto.cs]: look at down

File: VirtoCommerce.Storefront/Domain/Security/GraphQlSecurityHelper.cs (hunk 4 above)
Comment (kostyrin) [reply VirtoCommerce.Storefront/Domain/Security/GraphQlSecurityHelper.cs]: fixed

File: VirtoCommerce.Storefront/Domain/Security/GraphQlSecurityHelper.cs (hunk 5 above)
Comment (kostyrin) [reply VirtoCommerce.Storefront/Domain/Security/GraphQlSecurityHelper.cs:32]: image

File: VirtoCommerce.Storefront.Model/Security/Contracts/UpdateUserResponseDto.cs (hunk 3 above)
Comment (eMazeika) [reply VirtoCommerce.Storefront.Model/Security/Contracts/UpdateUserResponseDto.cs]: yes, and there is alias:
image

File: VirtoCommerce.Storefront/Domain/Security/GraphQlSecurityHelper.cs (hunk 5 above)
Comment (eMazeika) [reply VirtoCommerce.Storefront/Domain/Security/GraphQlSecurityHelper.cs]: yes, add the alias to graphql query and make sure to reuse existing SecurityResultResponseDto

**Pull request merged**